app/server.go

```diff
@@ -483,8 +483,15 @@ func NewServer(options ...Option) (*Server, error) {
 	} else {
 		nErr := backend.TestConnection()
 		if nErr != nil {
+			if errors.Is(nErr, filesstore.ErrNoS3Bucket) {
+				nErr = backend.(*filesstore.S3FileBackend).MakeBucket()
+				if nErr == nil {
+					goto backendTestSucceeded
+				}
+			}
 			mlog.Error("Problem with file storage settings", mlog.Err(nErr))
 		}
```

Comment:
The goto statement feels a bit unnecessary. We do run into some extra indentation, but it still keeps the code readable IMO.

```go
nErr := backend.TestConnection()
if nErr != nil {
  if errors.Is(nErr, filesstore.ErrNoS3Bucket) {
    mlog.Warn("Bucket specified does not exist. Attempting to create...")
    nErr = backend.(*filesstore.S3FileBackend).MakeBucket()
  }
  if nErr != nil {
    mlog.Error("Problem with file storage settings", mlog.Err(nErr))
  }
}
```